lib/portus/background/sync.rb

```diff
@@ -6,23 +6,45 @@ module Portus
   module Background
     # Sync synchronizes the contents from the registry into the DB.
     class Sync
+      def initialize
+        @executed = false
+      end
+
       # Returns how many seconds has to pass between each loop for this
       # synchronization to happen.
       def sleep_value
         20
       end
 
-      # Returns always true because this does not depend on some configuration
-      # options.
       def work?
-        true
+        return false unless APP_CONFIG.enabled?("background.sync")
+
+        val = APP_CONFIG["background"]["sync"]["sync-strategy"]
+        case val
+        when "update-delete", "update"
+          true
+        when "on-start"
+          !@executed
+        when "initial"
+          !@executed && !Repository.any?
```

Comment:
Does `empty?` work directly in the model? I tried in rails console but an exception happened.

Reply:
It doesn't. `empty?` is only usable on collections: e.g. `Repository.all.empty?`